src/test/java/com/redhat/darcy/webdriver/WebDriverCookieManagerAddGetDeleteCookieTest.java

```diff
@@ -0,0 +1,106 @@
+/*
+ Copyright 2014 Red Hat, Inc. and/or its affiliates.
+
+ This file is part of darcy-webdriver.
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.redhat.darcy.webdriver;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.redhat.darcy.web.Cookie;
+import com.redhat.darcy.webdriver.internal.TargetedWebDriver;
+import com.redhat.darcy.webdriver.internal.WebDriverCookieManager;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+import org.junit.runners.Parameterized.Parameter;
+import org.openqa.selenium.WebDriver;
+
+import java.time.LocalDateTime;
+import java.util.Arrays;
+import java.util.Collection;
+
+@RunWith(Parameterized.class)
+public class WebDriverCookieManagerAddGetDeleteCookieTest {
+
+    private static final Object[][] parameterizedChars = {
+            {new Cookie("chocolate", "chip")},
+            {new Cookie("chocolate", "chip", "")},
+            {new Cookie("chocolate", "chip", "", LocalDateTime.now().plusDays(1))},
+            {new Cookie("chocolate", "chip", "", null)},
+            {new Cookie("chocolate", "chip", null, "", LocalDateTime.now().plusDays(1))},
+            {new Cookie("chocolate", "chip", null, "", null, false)},
+            {new Cookie("chocolate", "chip", null, "", null, false, false)},
+    };
+
+    @Parameter
+    public Cookie cookie;
+
+    @Parameters
+    public static Collection<Object[]> data() {
+        return Arrays.asList(parameterizedChars);
+    }
+
+    @Test
+    public void shouldAddCookie() {
+        TargetedWebDriver mockDriver = mock(TargetedWebDriver.class);
+        WebDriver.Options mockOptions = mock(WebDriver.Options.class);
+
+        when(mockDriver.manage())
+                .thenReturn(mockOptions);
+
+        WebDriverCookieManager cookieManager = new WebDriverCookieManager(mockDriver);
+        cookieManager.add(cookie);
+
+        verify(mockOptions)
+                .addCookie(new org.openqa.selenium.Cookie(cookie.getName(), cookie.getValue()));
```

Comment:
I'm not sure the parameterized strategy is good to go here, unless you also parameterize the Selenium cookie that should be added. Essentially, you are not testing that the other parameters are correctly being translated to a Selenium cookie. The implementation could just pass `null` for everything but name and value and many/all of these tests would still pass.

So, I would prefer either parameterizing the cookie that should be expected as well as the cookie that is used as input and naming them in nice happy ways, or just have a test for each cookie constructor (named appropriately). In that case, I would combine this with the other test class since there would be no need for a separate runner.

In general, while parameterizing is more terse, it is easy for it to be less readable / more complicated.